test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -223,7 +223,15 @@ json_spirit::mObject fillBCTest(json_spirit::mObject const& _input)
 	TestBlock genesisBlock(_input.at("genesisBlockHeader").get_obj(), _input.at("pre").get_obj());
 	genesisBlock.setBlockHeader(genesisBlock.blockHeader());
 
-	TestBlockChain testChain(genesisBlock);
+    TestBlockChain::MiningType miningType = TestBlockChain::MiningType::Ethash;
+    if (_input.count("sealEngine") > 0)
+    {
+        output["sealEngine"] = _input.at("sealEngine");
+        if (_input.at("sealEngine") == eth::NoProof::name())
+            miningType = TestBlockChain::MiningType::NoProof;
+    }
+
+    TestBlockChain testChain(genesisBlock, miningType);
```

Comment:
if this exists only for assert below, you could do `assert(TestBlockChain(genesisBlock, miningType).getInterface().isKnown(...));`

Reply:
no testChain is required for BC tests logic